### PR TITLE
fix(AutoLayout): fixing issue when spacing was superior to padding causing a invalid GridLenght.

### PR DESCRIPTION
**Pull request by Robert-Louis**
GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

- When spacing were superior to padding it could cause the control to not work properly 

## What is the new behavior?

- Spacing and padding should behave as expeted

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.cs [hunk 1]
```diff
@@ -419,12 +424,17 @@ private void UpdateAlignments()
...
+					if (padding.Left - spacing > 0)
+					{
+						_grid.ColumnDefinitions.Insert(gridIndexOffSet, new ColumnDefinition() { Width = new GridLength(padding.Left - spacing) });
+						gridIndexOffSet += 1;
+					}
+
+					_grid.RowDefinitions.Add(new RowDefinition() { Height = new GridLength(padding.Top) });
+					_grid.RowDefinitions.Add(new RowDefinition() { Height = new GridLength(1, GridUnitType.Star) });
+					_grid.RowDefinitions.Add(new RowDefinition() { Height = new GridLength(padding.Bottom) }); 
 				}
```
Comment (carldebilly) [inline src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.cs:438]: Need a else here to clear padding. (+ same for vertical)

**Review by carldebilly — APPROVED** [review]

File: src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.cs (hunk 1 above)
Comment (Robert-Louis) [reply src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.cs:438]: The padding is reset each time according to the relation padding / spacing.

File: src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.cs [hunk 2]
```diff
@@ -393,9 +398,9 @@ private void UpdateAlignments()
 					//We need to make sure that the independent layout can span all across his parent
 					_grid.RowDefinitions.Add(new RowDefinition() { Height = new GridLength(1, GridUnitType.Star) });
 				}
-				if (hasPadding)
+				if (hasPadding && padding.Bottom - spacing > 0)
```
Comment (jeromelaban) [inline src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.cs:401]: I don't see a runtime test :) You should be able to build it now.

Comment (Robert-Louis) [reply src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.cs:401]: I will add one in the next commit to unblock Isaac as soon as possible.

**Pull request merged**